example/src/screens/Room.tsx

```diff
@@ -94,50 +106,78 @@ export const Room = ({ navigation }: Props) => {
+
+            {focusedParticipant ? (
+              <View style={styles.otherParticipants}>
+                <NotFocusedParticipants
+                  participants={participants.filter(
+                    (p) => p.id !== focusedParticipant?.id
+                  )}
+                />
+              </View>
+            ) : (
+              <View style={styles.participantsContainer}>
+                <View
+                  style={[
+                    styles.inner,
+                    participants.length > FLEX_BRAKPOINT
+                      ? styles.row
+                      : styles.column,
+                  ]}
+                >
+                  {participants
+                    .slice(
+                      0,
+                      participants.length > MAX_NUM_OF_USERS_ON_THE_SCREEN
+                        ? MAX_NUM_OF_USERS_ON_THE_SCREEN - 1
+                        : MAX_NUM_OF_USERS_ON_THE_SCREEN
+                    )
+                    .map((p) => (
+                      <View
+                        key={p.id}
+                        style={[
+                          getStylesForParticipants(participants),
+                          styles.shownParticipantBorder,
+                        ]}
+                      >
+                        <RoomParticipant
+                          participant={p}
+                          onPinButtonPressed={setFocusedParticipantId}
+                          titleSmall={participants.length > FLEX_BRAKPOINT}
+                        />
+                      </View>
+                    ))}
+
+                  {participants.length > MAX_NUM_OF_USERS_ON_THE_SCREEN && (
+                    <View style={getStylesForParticipants(participants)}>
+                      <OtherParticipants
+                        p1={participants[participants.length - 1]}
+                        p2={participants[participants.length - 2]}
+                        numOfOtherParticipants={
+                          participants.length -
+                          MAX_NUM_OF_USERS_ON_THE_SCREEN +
+                          1
+                        }
+                      />
                     </View>
-                  ))}
-
-                {participants.length > MAX_NUM_OF_USERS_ON_THE_SCREEN && (
-                  <View style={getStylesForParticipants(participants)}>
-                    <OtherParticipants
-                      p1={participants[participants.length - 1]}
-                      p2={participants[participants.length - 2]}
-                      numOfOtherParticipants={
-                        participants.length - MAX_NUM_OF_USERS_ON_THE_SCREEN + 1
-                      }
-                    />
-                  </View>
-                )}
+                  )}
+                </View>
               </View>
-            </View>
+            )}
           </View>
-          <CallControls />
+          <CallControls bottomOffset={useSafeAreaInsets().bottom} />
```

Comment:
please use hooks before returning components
we're often rendering things conditionally and react is going to crash if you start/stop calling hook within lifetime of a component

Reply:
Done.